src/server/main_service.cc

```diff
@@ -590,6 +600,42 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, std::move(listeners), &cluster_family_);
 
+  if (string flag = GetFlag(FLAGS_replicaof); !flag.empty()) {
+    string ip, port;
+    {
+      auto pos = flag.find_last_of(':');
+      CHECK(pos != string::npos) << "Invalid format for --replicaof: missing ':'.";
+
+      string ip1 = flag.substr(0, pos);
+      port = flag.substr(pos + 1);
+
+      CHECK(!ip1.empty() && !port.empty())
+          << "Invalid format for --replicaof: IP or port are empty.";
+
+      // For IPv6
+      CHECK(!((ip1.front() == '[') ^ (ip1.back() == ']')))
+          << "Invalid format for --replicaof: unclosed brackets.";
+
+      if (ip1.front() == '[') {
+        ip = ip1.substr(1, ip1.length() - 2);
+        LOG(INFO) << "Received IPv6: " << ip;
```

Comment:
Given users will probably see this, I'd make it more verbose. Maybe "Replicating IPv6 << ip"? (also below)
"Received" might be confusing in the log for users

Reply:
And given the log in line 628, I'd make both of these `VLOG` instead of `INFO` level.